test/jdk/javax/swing/JSlider/8262981/JSliderAccessibleAction.java

```diff
@@ -0,0 +1,312 @@
+                            + jSliderInitialValue + "  currentJSliderValue = " + currentJSliderValue);
+                }
+
+                if (!actionDescriptionValue.get()) {
+                    throw new RuntimeException("Expected that JSlider's " +
+                            "AccessibleAction getAccessibleActionDescription(2) " +
+                            "return null");
+                }
+
+                mouseAction(robot, invalidIncrementBtn);
+                if (!invalidIncrementCountDownLatch.await(30,
+                        TimeUnit.SECONDS)) {
+                    throw new RuntimeException("Failed to perform action on Invalid " +
+                            "Increment button");
+                }
+                if (jSliderInitialValue.get() != currentJSliderValue.get()) {
+                    throw new RuntimeException("Expected that JSlider value is not " +
+                            "changed when invalid decrement value -1 is passed to " +
+                            "doAccessibleAction(-1)  jSliderInitialValue = "
+                            + jSliderInitialValue + "  currentJSliderValue = " + currentJSliderValue);
+                }
+
+                if (!actionDescriptionValue.get()) {
+                    throw new RuntimeException("Expected that JSlider's " +
+                            "AccessibleAction getAccessibleActionDescription(-1) " +
+                            "return null");
+                }
+
+                // JSlider value is decremented
+                mouseAction(robot, decrementBtn);
+                if (!validDecrementCountDownLatch.await(30, TimeUnit.SECONDS)) {
+                    throw new RuntimeException("Failed to perform action on valid " +
+                            "decrement button");
+                }
+                if (jSliderInitialValue.get() == currentJSliderValue.get()) {
+                    throw new RuntimeException("Expected that JSlider value is  " +
+                            "decremented when value 1 is passed to " +
+                            "doAccessibleAction(1)  jSliderInitialValue = "
+                            + jSliderInitialValue + "  currentJSliderValue = " + currentJSliderValue);
+                }
+
+                if (!actionDescriptionValue.get()) {
+                    throw new RuntimeException("Expected that JSlider's " +
+                            "AccessibleAction getAccessibleActionDescription(1) " +
+                            "return AccessibleAction.DECREMENT");
+                }
+
+                // JSlider value is incremented
+                mouseAction(robot, incrementBtn);
+                if (!validIncrementCountDownLatch.await(30, TimeUnit.SECONDS)) {
+                    throw new RuntimeException("Failed to perform action on valid " +
+                            "Increment button");
+                }
+                if (jSliderInitialValue.get() != currentJSliderValue.get()) {
```

Comment:
This is  fundamentally unstable. 

You are clicking on the decrease button. The actual decrease which happened in the slider after the click may be more than you expect. Or less. 

The only reliable condition perhaps is that the value has decreased and even that may have a delayed effect, so it needs to be waited for. 

For example see  sanity/client/SwingSet/src/SliderDemoTest.java.

Reply:
Fixed this by adding addChangeListener to JSlider and updating the currentJSliderValue.